server/bin/pbench-verify-backup-tarballs.py

```diff
@@ -0,0 +1,349 @@
+                i.name, backup_obj.description)
+            report.write(report_text)
+    elif i < len_s3_content and j == len_backup_content:
+        for i in sorted_s3_content[i:len_s3_content]:
+            report_text = "{}: present in s3 but not in {}\n".format(
+                i.name, backup_obj.description)
+            report.write(report_text)
+    return Status.SUCCESS
+
+
+def report_primary_backup_s3(archive_obj, backup_obj, s3_backup_obj, tmpdir, report, logger):
+    # Function to compare archive, backup and s3
+    aname = os.path.join(tmpdir, archive_obj.list_name)
+    try:
+        with open(aname) as f:
+            primary_list = set(f.readlines())
+    except Exception as e:
+        logger.exception("error reading {}, {}\n".format(aname, e))
+        return Status.FAIL
+
+    bname = os.path.join(tmpdir, backup_obj.list_name)
+    try:
+        with open(bname) as f:
+            backup_list = set(f.readlines())
+    except Exception as e:
+       logger.exception("error reading {}, {}\n".format(bname, e))
+       return Status.FAIL
+
+    sname = os.path.join(tmpdir, s3_backup_obj.list_name)
+    try:
+        with open(sname) as f:
+            s3_list = set(f.readlines())
+    except Exception as e:
+        logger.exception("error reading {}, {}\n".format(sname, e))
+        return Status.FAIL
+
+    only_p = primary_list.difference(backup_list, s3_list)
+    for i in sorted(only_p):
+        report.write("{}: only in archive\n".format(i.strip('\n')))
+
+    only_b = backup_list.difference(primary_list, s3_list)
+    for j in sorted(only_b):
+        report.write("{}: only in backup\n".format(j.strip('\n')))
+
+    only_s3 = s3_list.difference(primary_list, backup_list)
+    for k in sorted(only_s3):
+        report.write("{}: only in s3\n".format(k.strip('\n')))
+    return Status.SUCCESS
+
+
+def main():
+    cfg_name = os.environ.get("CONFIG")
+    if not cfg_name:
+        print("{}: ERROR: No config file specified; set CONFIG env variable or"
+                " use --config <file> on the command line".format(_NAME_),
+                file=sys.stderr)
+        return 2
+
+    try:
+        config = PbenchConfig(cfg_name)
+    except BadConfig as e:
+        print("{}: {}".format(_NAME_, e), file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    archive = config.ARCHIVE
+    if not os.path.isdir(archive):
+        logger.error(
+            'The ARCHIVE directory does not resolve to a directory, {}\n'.format(archive))
+        return 1
+
+    # add a BACKUP field to the config object
+    config.BACKUP = backup = config.conf.get("pbench-server", "pbench-backup-dir")
+    if len(backup) == 0:
+        logger.error(
+            'Unspecified backup directory, no pbench-backup-dir config in pbench-server section\n')
+        return 1
+
+    if not os.path.isdir(backup):
+        logger.error('Specified backup directory, {}, does not resolve {} to a directory\n'.format(
```

Comment:
This will produce something like: "Specifiied backup directory, /a/b/c, does resolve /d/e/f to a directory": if I'm reading the log file, how am I supposed to understand what this message means?